plugins/prometheus.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+from jinja2 import Template
+import json
+from matrix_client.api import MatrixRequestError
+from neb.engine import KeyValueStore, RoomContextStore
+from neb.plugins import Plugin, admin_only
+from Queue import PriorityQueue
+from threading import Thread
+
+
+import time
+import logging as log
+import random
+
+queue = PriorityQueue()
+
+
+class PrometheusPlugin(Plugin):
+    """Plugin for interacting with Prometheus.
+    """
+    name = "prometheus"
+
+    #Webhooks:
+        #    /neb/prometheus
+    TYPE_TRACK = "org.matrix.neb.plugin.prometheus.projects.tracking"
+
+    TRACKING = ["track", "tracking"]
+
+    def __init__(self, *args, **kwargs):
+        super(PrometheusPlugin, self).__init__(*args, **kwargs)
+        self.store = KeyValueStore("prometheus.json")
+        self.rooms = RoomContextStore(
+            [PrometheusPlugin.TYPE_TRACK]
+        )
+        self.queue_counter = 1L
```

Comment:
From what I can see, this is never incremented. As a result, I guess that means `PriorityQueue` degrades into a FIFO queue. This has implications with your retry scheduling (things pushed on `:89` will be at the back of the queue, whereas presumably we want it at the front of the queue).